### PR TITLE
#578 More extensive database seeding - bulk creating members, gardens and plantings

### PR DESCRIPTION
**Pull request by oshiho3**
Made some improvement on the database seeding process, covering some items from issue #578. Just did this as I wanted to test performance on geo related processing.

This will let you...
- Create N number of members and 
- Populate member's location
- Populate garden's location
- Creates a planting from various crops, sunniness and planted_from for each user.

The default number of members is still 3, but you can specify the number of members as...
<code>rake db:seed member_size=100</code>
or
<code>rake db:reset member_size=100<code>
Tested up to 1000 worked fine.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -60,16 +60,50 @@ def load_basic_account_types
...
+
+  (1..member_size).each do |i|
+    @user = Member.new(
         :login_name => "test#{i}",
         :email => "test#{i}@example.com",
         :password => "password#{i}",
         :tos_agreement => true
     )
-    @user.confirm!
+    @user.skip_confirmation!
```
Comment (cesy) [inline db/seeds.rb]: What's the rationale for this change?

File: db/seeds.rb [hunk 2]
```diff
@@ -60,16 +60,50 @@ def load_basic_account_types
...
+      # Using 'update_column' method instead of 'update' so that 
+      # it avoids accessing Geocoding service for faster processing
+      @user.gardens.first.update_columns(location: suburb, latitude: latitude, longitude: longitude)
+      @user.update_columns(location: suburb, latitude: latitude, longitude: longitude)
+    end
+
+    # Create a planting by the member
+    Planting.create(
+      owner_id: @user.id,
+      garden_id: @user.gardens.first.id,
```
Comment (cesy) [inline db/seeds.rb]: Where do we create the gardens? It doesn't look like this will auto-create.

File: db/seeds.rb (hunk 1 above)
Comment (oshiho3) [inline db/seeds.rb]: Thanks for looking at my code. Currently when a test user is loaded, content of the confirmation email for each user opens up in a browser. However, <code>skip_confirmation</code> automatically confirm the user and avoid generating thousands of confirmation emails/pages. My rationale is most of the time we ignore the confirmation pages, and if someone wants to test the confirmation email content specifically, it can be tested manually.

Comment (maco) [inline db/seeds.rb]: I find those browser windows annoying; therefore, I love you right now.

File: db/seeds.rb (hunk 2 above)
Comment (oshiho3) [inline db/seeds.rb]: To be honest, I was puzzled myself for a moment when I looked at it now. It's actually done in the Member model. It creates a default garden when a member is created, so in the test script, I just needed to populate the location.

File: db/seeds.rb (hunk 1 above)
Comment (oshiho3) [inline db/seeds.rb]: Thanks for your endorsement :-) yes, it was annoying me too.

Comment (Skud) [inline db/seeds.rb]: This is great, thanks! I suspect .skip_confirmation is new since we 
started using Devise but it is a very nice thing to be able to use, and 
avoid those popup windows.

On 26/03/2015 8:08 am, Shiho Takagi wrote:
>
> In db/seeds.rb 
> <https://github.com/Growstuff/growstuff/pull/731#discussion_r27166170>:
>
> >          :login_name => "test#{i}",
> >          :email => "test#{i}@example.com",
> >          :password => "password#{i}",
> >          :tos_agreement => true
> >      )
> > -    @user.confirm!
> > +    @user.skip_confirmation!
>
> Thanks for looking at my code. Currently when a test user is loaded, 
> content of the confirmation email for each user opens up in a browser. 
> However, |skip_confirmation| automatically confirm the user and avoid 
> generating thousands of confirmation emails/pages. My rationale is 
> most of the time we ignore the confirmation pages, and if someone 
> wants to test the confirmation email content specifically, it can be 
> tested manually.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Growstuff/growstuff/pull/731/files#r27166170>.
>

-- 
Alex "Skud" Bayley
skud@growstuff.org
http://growstuff.org/

**Comment by oshiho3** [comment]
Added sunniness and planted_from field data population on plantings.
Also made all the user confirmation email skipped as some said they also don't want to see the page popping up.

**Comment by oshiho3** [comment]
Added sunniness and planted_from field on planting.
Also got rid of all the user confirmation popup since some said they like the idea.

**Comment by Marlena** [comment]
This will help move work forward on the pull request for sunniness and planting charts.  Hope we can merge this one in soon.

**Comment by maco** [comment]
As soon as release 8 is pushed live, I have about 10 things to merge right
away.

**Comment by Marlena** [comment]
That's cool.  I can see that there is a large queue.  Working on pulling in the branch and just using that for now.

**Comment by Skud** [comment]
This looks great and I'm going to merge it :)  I would support the idea of automatically creating 20 or so test users, as a default case, fwiw.

**Comment by oshiho3** [comment]
Welcome back, @Skud! Yup, increasing the number of default users sounds good. I think I'll implement it in the next iteration along with other improvements in database seeding.
